tpr/tpr.go

```diff
@@ -22,48 +28,74 @@ const (
 // Config is a TPR configuration.
 type Config struct {
 	// Dependencies.
-	Clientset kubernetes.Interface
+	K8sClient            kubernetes.Interface
+	Logger               micrologger.Logger
+	ResourceEventHandler cache.ResourceEventHandler
+	ZeroObjectFactory    ZeroObjectFactory
```

Comment:
What if I want to get an `Endpoint` but not create `NewInformer`?
I'd like these two guys to be params of `NewInformer`.